### PR TITLE
exporting JBIG2 images

### PR DESCRIPTION
**Pull request by side2k**
This PR is adaptation of this one: https://github.com/euske/pdfminer/pull/107

### CODE REVIEW

**Comment by vstoykov** [comment]
There is no `cStringIO` in Python2.

**Comment by goulu** [comment]
Please check why tests fail and fix this before we can merge...
Thank you !

**Comment by vstoykov** [comment]
@side2k you can replace `input_stream = StringIO()` with `input_stream = BytesIO()` (`BytesIO` is already imported) and remove the import of `StringIO`. Also you can rebase probably.

**Comment by pietermarsman** [comment]
This PR needs only a little work. @side2k can you do that?

**Comment by pietermarsman** [comment]
This PR fixes #26

**Comment by side2k** [comment]
@pietermarsman will look into it within a nearest couple of hours

**Comment by side2k** [comment]
@pietermarsman i've rebased and added the fixes. All the tests are passing now.

**Comment by side2k** [comment]
it would be really great to get a code review from someone who is familiar with the current state of things in pdfminer - I didn't touch it for a couple of years by now.

**Comment by pietermarsman** [comment]
Do you have a pdf and script to test it with?

**Comment by pietermarsman** [comment]
(@side2k , not sure if you've missed my previous message, this is a friendly notification if you did)

Do you have a pdf and sample code to test this with? I can't understand and check this PR if I don't have a pdf that triggers this code.

**Comment by side2k** [comment]
@pietermarsman I think I had once, but right now I dont have an opportunity to search. Maybe later. I am on a business trip right now, sorry.

**Comment by pietermarsman** [comment]
@side2k let us know when you find it. 

I think we can close this PR until we have some testing material. Once we have that we can reopen, test, review and merge.

**Comment by pietermarsman** [comment]
@ganeshtata, there is no pdf to test this PR with. Do you agree that we cannot merge this PR if there is nothing to test?

File: pdfminer/jbig2.py [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+
+# segment types
+
+SEG_TYPE_IMMEDIATE_GEN_REGION = 38
+SEG_TYPE_END_OF_PAGE = 49
+SEG_TYPE_END_OF_FILE = 50
+
+# file literals
+
+FILE_HEADER_ID = '\x97\x4A\x42\x32\x0D\x0A\x1A\x0A'
```
Comment (pietermarsman) [inline pdfminer/jbig2.py]: Should be a bytestring

File: pdfminer/jbig2.py [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+        if not check_flag(FILE_HEAD_FLAG_PAGES_UNKNOWN, file_flags):
+            page_count = unpack(">L", self.stream.read(4))
+
+        # reading segments
+        segments = self.get_segments()
+
+        return segments
+
+    def is_eof(self):
+        if self.stream.read(1) == '':
```
Comment (pietermarsman) [inline pdfminer/jbig2.py]: Should compare to a bytestring, e.g. `self.stream.read(1) == b''`

File: pdfminer/jbig2.py [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+        eof_segment = self.get_eof_segment(seg_num+1)
+        data = self.encode_segment(eof_segment)
+
+        self.stream.write(data)
+        data_len += len(data)
+
+        return data_len
+
+    def encode_segment(self, segment):
+        data = ''
```
Comment (pietermarsman) [inline pdfminer/jbig2.py]: Should be a bytestring

File: pdfminer/jbig2.py [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+            }
+        }
+
+    def get_eof_segment(self, seg_number):
+        return {
+            'data_length': 0,
+            'flags': {'deferred': False, 'type': SEG_TYPE_END_OF_FILE},
+            'number': seg_number,
+            'page_assoc': 0,
+            'raw_data': '',
```
Comment (pietermarsman) [inline pdfminer/jbig2.py]: Should be a bytestring

**Comment by pietermarsman** [comment]
I've found a pdf with JBIG2 images [pdfbox jira](https://issues.apache.org/jira/browse/PDFBOX-1067).

[ItDoesntWorkScan.pdf](https://github.com/pdfminer/pdfminer.six/files/3754470/ItDoesntWorkScan.pdf)

**Pull request closed without merging**